src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,216 @@
+    [
+      modal,
+      size === Small && modalSm,
+      size === Large && modalLg,
+      size === FullScreen && modalFs,
+    ]
+  ), [size])
+  const modalBodyClassNames = useMemo(() => classNames(
+    [
+      body,
+      isBodyFullWidth && bodyFullWidth,
+    ]
+  ), [isBodyFullWidth])
+  const modalContentWrapperClassNames = useMemo(() => classNames(
+    [
+      content,
+      size !== Small && extension && contentWrapper,
+    ]
+  ), [extension, size])
+
+  const [isExtensionOpen, setIsExtensionOpen] = useState(false)
+
+  const modalTitle = useMemo(() => (
+    <div className={styles.title}>
+      {title && <>
+        <H4 ellipsis={{ rows: 1, tooltip: title }}>{title}</H4>
+        {docLink && <div className={styles.docLink}>
+          <Button
+            icon={docLinkIcon}
+            shape={Circle}
+            type={Ghost}
+            onClick={onClickDocLink}
+          />
+        </div>}
+      </>}
+    </div>
+  ), [docLink, docLinkIcon, onClickDocLink, title])
+
+  const modalFooter = useMemo(() => {
+    const { buttons, extra } = footer || {}
+    return (
+      <div className={styles.footer}>
+        {footer && <>
+          <div className={footerButtons}>
+            {buttons}
+          </div>
+          {extra}
+        </>}
+      </div>
+    )
+  }, [footer])
+
+  const modalContent = useMemo(() => {
+    if (size !== Small && extension) {
+      const {
+        children: extChildren,
+        isFixed,
+        labelClose,
+        labelOpen,
+        title: extTitle,
+      } = extension || {}
+
+      const modalExtension = (
+        <div className={styles.extension}>
+          <BodyM>{extTitle}</BodyM>
+          {extChildren}
+        </div>
+      )
+
+      if (isFixed) {
+        return (
+          <>
+            <div className={content}>
+              {children}
+            </div>
+            {modalExtension}
+          </>
+        )
+      }
+
+      const changeExtensionStatus = (): void => setIsExtensionOpen(!isExtensionOpen)
+
+      const modalExtensionLabel = (
+        <div className={extensionLabel}>
+          <Button
+            icon={<Icon color="currentColor" name={isExtensionOpen ? 'PiCaretLeft' : 'PiCaretRight'} size={16} />}
+            iconPosition={isExtensionOpen ? Left : Right}
+            type={Ghost}
+            onClick={changeExtensionStatus}
+          >
+            {isExtensionOpen ? labelClose : labelOpen}
+          </Button>
+        </div>
+      )
+
+      return (
+        <>
+          <div className={modalContentWrapperClassNames}>
+            <div className={content}>
+              {children}
+            </div>
+            {modalExtensionLabel}
+          </div>
+          {isExtensionOpen && modalExtension}
+        </>
+      )
+    }
+
+    return (
+      <div className={content}>
+        {children}
+      </div>
+    )
+  }, [children, extension, isExtensionOpen, modalContentWrapperClassNames, size])
```

Comment:
il children è una dipendenza che cambia sempre no?

Reply:
Not necessarily, we have modals whose content is only informative. In addition, components such as Ant's Input or Form do not generate a change of `children` upon, for example, text entry or the flag of a checkbox